### PR TITLE
Closes #2183, #2184, #2185: add "reply to" for notification e-mails

### LINKED ISSUE #2183: Modify "Return for modification" modal window
image

Under "Additional information" add input field labeled "Return e-mail address"/"Zwrotny adres e-mail". After the label there should be a question mark icon triggering popup with the following message: 

EN:
An email address for contact regarding review of this dataset. This address will be made available to the person depositing this dataset and should allow further contact with the person returning it. 

PL: 
Adres e-mail służący do kontaktu w sprawie weryfikacji tego zbioru danych. Adres ten zostanie udostępniony osobie deponującej zbiór i powinien umożliwiać dalszy kontakt z osobą zwracającą zbiór. 

* The field should be required. 
* The field should be pre-filled with the e-mail of the user returning the dataset. 
* The field should be checked if it contains a valid e-mail address (here we can mimic "Email Dataverse Contact" modal). 

We can also remove colon after "Additional information" - its obsolete.

### PR DESCRIPTION
**Pull request by wtr-icm**

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/notification/UserNotificationService.java [hunk 1]
```diff
@@ -51,91 +62,115 @@ public UserNotificationService(UserNotificationRepository userNotificationReposi
...
+//     * @param notificationObjectType - type has to match correct #{@link NotificationType}
+//     */
+//    public void sendNotificationWithEmail(AuthenticatedUser dataverseUser,
+//                                          Timestamp sendDate,
+//                                          String type,
+//                                          Long dvObjectId,
+//                                          NotificationObjectType notificationObjectType,
+//                                          AuthenticatedUser requestor) {
+//
+//        UserNotification userNotification = createUserNotification(dataverseUser, sendDate, type, dvObjectId, requestor);
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/notification/UserNotificationService.java]: Can we tidy up commented code in this class?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/notification/UserNotificationService.java [hunk 2]
```diff
@@ -27,20 +34,24 @@ public class UserNotificationService {
     private UserNotificationRepository userNotificationRepository;
     private MailService mailService;
     private EmailNotificationMapper mailMapper;
+    private ObjectMapper objectMapper;
+    private TypeReference<HashMap<String, String>> parametersTypeRef = new TypeReference<HashMap<String, String>>() {};
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/notification/UserNotificationService.java]: Is it possible to make this field static?

File: dataverse-webapp/src/main/resources/config/dataverse.default.properties [hunk 3]
```diff
@@ -15,7 +15,7 @@ Languages=[{"locale":"en", "title":"English"}, {"locale":"pl", "title":"Polski"}
 DefaultDateFormat=yyyy-MM-dd
 
 AllRightsReservedTermsOfUseActive=false
-RestrictedAccessTermsOfUseActive=false
+RestrictedAccessTermsOfUseActive=true
```
Comment (madryk) [inline dataverse-webapp/src/main/resources/config/dataverse.default.properties]: Did we do that on purpouse?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/SubmitDatasetForReviewCommand.java [hunk 4]
```diff
@@ -61,21 +63,26 @@ public Dataset save(CommandContext ctxt)  {
...
     }
 
+    private void sendNotification(CommandContext ctxt, Dataset savedDataset) {
+        User user = getUser();
+        if (user == null || !user.isAuthenticated()) {
+            return;
+        }
+                List<AuthenticatedUser> curators = ctxt.permissions().getUsersWithPermissionOn(Permission.PublishDataset, savedDataset);
+        AuthenticatedUser requestor = (AuthenticatedUser) user;
+        Timestamp timestamp = new Timestamp(new Date().getTime());
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/SubmitDatasetForReviewCommand.java]: Little problem with indentation

File: dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/user/UserNotification.java [hunk 5]
```diff
@@ -31,98 +32,117 @@ public class UserNotification implements Serializable, JpaEntity<Long> {
     @ManyToOne
     @JoinColumn(nullable = false)
     private AuthenticatedUser user;
+
     @ManyToOne
     @JoinColumn(nullable = true)
     private AuthenticatedUser requestor;
```
Comment (madryk) [inline dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/user/UserNotification.java:38]: Can we create a task to get rid of this and `additionalMessage`?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/mail/MailMessageCreator.java [hunk 6]
```diff
@@ -69,22 +68,24 @@ public MailMessageCreator() {
...
 
     /**
      * Creates footer for email message.
      */
-    public String createMailFooterMessage(Locale messageLocale, String rootDataverseName, InternetAddress systemAddress) {
-
-        return BundleUtil.getStringFromBundleWithLocale("notification.email.closing", messageLocale,
+    public String createMailFooterMessage(String notificationType, Locale messageLocale, String rootDataverseName, InternetAddress systemAddress) {
+        String closingLine = RETURNEDDS.equals(notificationType)
+                ? "notification.email.closing.other" : "notification.email.closing";
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/mail/MailMessageCreator.java]: I think that it is something wrong that you have to pass `notificationType` here. From what I understand some of the mails sent doesn't even have "notification nature". I would try maybe to leave the method as it was - maybe with renaming it to `createMailDefaultFooterMessage` and somewhere in higher level when we construct mail text for specific notification make a distinction

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/notification/UserNotificationService.java [hunk 7]
```diff
@@ -51,91 +62,115 @@ public UserNotificationService(UserNotificationRepository userNotificationReposi
...
-     */
-    public void sendNotificationWithEmail(AuthenticatedUser dataverseUser,
-                                          Timestamp sendDate,
-                                          String type,
-                                          Long dvObjectId,
-                                          NotificationObjectType notificationObjectType,
-                                          String comment) {
+    public void setParameters(UserNotification notification, Map<String, String> parameters) {
+        notification.setParameters(parametersToString(parameters));
+    }
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/notification/UserNotificationService.java]: IMO these two methods do not quite belong to this service. IMO they pollute public api of the service. I would try to extract it to some either some other service or make a util class. But I'm not insisting on it. We can probably live with this

**Review by madryk — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/resources/config/dataverse.default.properties (hunk 3 above)
Comment (wtr-icm) [reply dataverse-webapp/src/main/resources/config/dataverse.default.properties]: Accident :)

File: dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/user/UserNotification.java (hunk 5 above)
Comment (wtr-icm) [reply dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/user/UserNotification.java:38]: https://github.com/CeON/dataverse/issues/2202

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/mail/MailMessageCreator.java (hunk 6 above)
Comment (wtr-icm) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/mail/MailMessageCreator.java]: I've moved that code one level higher, and also I've made a second version with `null` as notification type in order to hide that `null` in other places.

**Review by madryk — APPROVED** [review]

**Pull request merged**